spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/hazelcast/HazelcastHealthIndicator.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.hazelcast;
+
+import java.net.InetSocketAddress;
+import java.net.UnknownHostException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import com.hazelcast.core.Client;
+import com.hazelcast.core.Endpoint;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.Member;
+import com.hazelcast.nio.Address;
+
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.util.Assert;
+
+/**
+ * {@link HealthIndicator} for a Hazelcast.
+ *
+ * @author Dmytro Nosan
+ * @since 2.2.0
+ */
+public class HazelcastHealthIndicator extends AbstractHealthIndicator {
+
+	private final HazelcastInstance hazelcast;
+
+	public HazelcastHealthIndicator(HazelcastInstance hazelcast) {
+		super("Hazelcast health check failed");
+		Assert.notNull(hazelcast, "HazelcastInstance must not be null");
+		this.hazelcast = hazelcast;
+	}
+
+	@Override
+	protected void doHealthCheck(Health.Builder builder) {
+		HazelcastInstance hazelcast = this.hazelcast;
+		hazelcast.executeTransaction((context) -> builder.up().withDetails(getDetails(hazelcast)));
+	}
+
+	private static Map<String, Object> getDetails(HazelcastInstance hazelcast) {
+		Map<String, Object> details = new LinkedHashMap<>();
+		details.put("name", hazelcast.getName());
+		Endpoint endpoint = hazelcast.getLocalEndpoint();
+		if (endpoint instanceof Member) {
```

Comment:
I would stick to `name` and `uuid` for now. Not denying that extra information might be useful but I'd start with a lighter surface area for a start.

Reply:
Should we still split the details between `client` and `member`?

Reply:
IMO not at this time, no.